File: ios_app/GenAIAcademyDashboard/Models/DashboardSummary.swift
//
//  DashboardSummary.swift
//  GenAIAcademyDashboard
//
//  Dashboard summary model matching backend API response
//

import Foundation

struct DashboardSummary: Codable {
    let totalUsers: Int
    let uniqueOrganizations: Int
    let topDomain: String
    let topCity: String
    let topOrganization: String?
    let uniqueCountries: Int?
    let usersWithGithub: Int?
    let usersWithLinkedin: Int?
    let averageAge: Int?
    
    enum CodingKeys: String, CodingKey {
        case totalUsers = "total_users"
        case uniqueOrganizations = "unique_organizations"
        case topDomain = "top_domain"
        case topCity = "top_city"
        case topOrganization = "top_organization"
        case uniqueCountries = "unique_countries"
        case usersWithGithub = "users_with_github"
        case usersWithLinkedin = "users_with_linkedin"
        case averageAge = "average_age"
    }
}

File: flutter_app/lib/models/dashboard_summary.dart
import 'package:json_annotation/json_annotation.dart';

part 'dashboard_summary.g.dart';

@JsonSerializable()
class DashboardSummary {
  @JsonKey(name: 'total_users')
  final int totalUsers;
  
  @JsonKey(name: 'unique_organizations')
  final int uniqueOrganizations;
  
  @JsonKey(name: 'top_domain')
  final String topDomain;
  
  @JsonKey(name: 'top_city')
  final String topCity;
  
  @JsonKey(name: 'top_organization')
  final String? topOrganization;
  
  @JsonKey(name: 'unique_countries')
  final int? uniqueCountries;
  
  @JsonKey(name: 'users_with_github')
  final int? usersWithGithub;
  
  @JsonKey(name: 'users_with_linkedin')
  final int? usersWithLinkedin;
  
  @JsonKey(name: 'average_age')
  final int? averageAge;
  
  DashboardSummary({
    required this.totalUsers,
    required this.uniqueOrganizations,
    required this.topDomain,
    required this.topCity,
    this.topOrganization,
    this.uniqueCountries,
    this.usersWithGithub,
    this.usersWithLinkedin,
    this.averageAge,
  });
  
  factory DashboardSummary.fromJson(Map<String, dynamic> json) => _$DashboardSummaryFromJson(json);
  Map<String, dynamic> toJson() => _$DashboardSummaryToJson(this);
}
